File: BitManipulation/C_PowersOfTwo.cpp
#include <bits/stdc++.h>
using namespace std ;
int main() {
	int n , k ;
	scanf("%d %d", &n , &k);
	vector<int> ans(k , 1) ;
	n -= k ;
	for (int i = 0 ; n && i < k ; i++) {
		while (ans[i] <= n) {
			n -= ans[i];
			ans[i] *= 2;
		}
	}
	if (n) puts("NO") ;
	else {
		puts("YES") ;
		for (auto u : ans) printf("%d ", u);
		puts("");
	}

}

File: BitManipulation/C_PowersOfTwo.py
n, k = map(int, input().split())
a = [1]*k
n -= k
i = 0
while i < k and n:
    while a[i] <= n:
        n -= a[i]
        a[i] *= 2
        i += 1
        if n:
            print("NO")
        else:
            print("YES", *a)
